app/angular/src/client/preview/angular-beta/ComputesTemplateFromComponent.ts

```diff
@@ -127,5 +128,55 @@ export const computesTemplateSourceFromComponent = (
       ? ` ${initialOutputs.map((i) => `(${i})="${i}($event)"`).join(' ')}`
       : '';
 
-  return `<${ngComponentMetadata.selector}${templateInputs}${templateOutputs}></${ngComponentMetadata.selector}>`;
+  const template = buildTemplate(ngComponentMetadata.selector);
+  return `<${template.openTag}${templateInputs}${templateOutputs}></${template.closeTag}>`;
+};
+
+const buildTemplate = (
+  selector: string
+): {
+  openTag?: string;
+  closeTag?: string;
+} => {
+  const templates = [
+    {
+      // Match element selectors with optional chained attributes or classes
+      re: /^([\w\d-_]+)(?:(?:\[([\w\d-_]+)(?:=(.+))?\])|\.([\w\d-_]+))?/,
```

Comment:
a long time ago when I was doing Jquery I used this solution (regex) : https://codegolf.stackexchange.com/a/123281
I don't know if you made them yourself but maybe this could improve or simplify 🤷‍♂️

Reply:
Done with modifications